### PR TITLE
[C-2391 PLAT-833] Handle get-discovery-notification error, notification processing

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Fixes a few error cases related to new discovery notifications.
- Handles case where api returns null
- Handles case where `mapNotifications` wasn't truly transforming `DiscoveryNotification` to `Notification`
- Fixes processNotification function to include id data from new notifs.

### CODE REVIEW

File: packages/web/src/common/store/notifications/parseAndProcessNotifications.ts [hunk 1]
```diff
@@ -85,58 +85,65 @@ export function* parseAndProcessNotifications(
...
       )
     }
-    if (notification.type === NotificationType.RemixCreate) {
+    if (type === NotificationType.RemixCreate) {
       trackIdsToFetch.push(
         notification.parentTrackId,
         notification.childTrackId
       )
       notification.entityType = Entity.Track
-      notification.entityIds = [
```
Comment (sabrina-kiam) [inline packages/web/src/common/store/notifications/parseAndProcessNotifications.ts:104]: just confirming this is no longer needed?

**Review by sabrina-kiam — APPROVED** [review]
thanks for cleaning up as well as fixing! so should this fix clicking share on tastemaker and it disappearing as well as the feed issue?

File: packages/web/src/common/store/notifications/parseAndProcessNotifications.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/web/src/common/store/notifications/parseAndProcessNotifications.ts:104]: yes! it was a mistyped notif

**Comment by dylanjeffers** [comment]
> thanks for cleaning up as well as fixing! so should this fix clicking share on tastemaker and it disappearing as well as the feed issue?

Yes! since the issue was due to the user not being there basically

**Pull request merged**